components/get_post_link_modal/get_post_link_modal.tsx

```diff
@@ -2,57 +2,73 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
 
 import ModalStore from 'stores/modal_store.jsx';
 import Constants from 'utils/constants';
 import * as Utils from 'utils/utils.jsx';
 import GetLinkModal from 'components/get_link_modal';
 
-export default class GetPostLinkModal extends React.PureComponent {
-    static propTypes = {
+type Props = {
+    currentTeamUrl: string;
+}
 
-        /**
-         * URL of current team
-         */
-        currentTeamUrl: PropTypes.string.isRequired,
+type State = {
+    show: boolean;
+    post: {
```

Comment:
Not related to the migration itself, but I think we can improve it here. Looks like the post object is only using the post id, but we are saving the whole post in the state, maybe we can just have a `postId` field in the State.